Remove references to projectlogging and clusterlogging CRDs

JonCrowther opened the pull request:
## Issue: https://github.com/rancher/rancher/issues/47972
 
## Problem
The CRDs `projectlogging.management.cattle.io` and `clusterlogging.management.cattle.io` haven't existed for a long time in Rancher, but they still exist in the code base.
 
## Solution
Remove the references to `clusterlogging` and `projectlogging` from the code. 

I am also removing the password store from Norman. From the PR that added the store https://github.com/rancher/rancher/pull/15968

>    `password/store.go` contains generic logic to create secret for any field that is of type password. To add a schema that has password fields, it needs to be added to `pwdTypes`.

The only schemas in password/store.go are clusterlogging and projectlogging, so the store is a no-op if those CRDs are gone.
 
## Testing
When installing Rancher, neither CRD is installed, and there are numerous parts of the code that attempt to remove the namespace that was used for this old logging system `cattle-logging`. 

## Engineering Testing
### Manual Testing
Manual testing was building and running Rancher and watching the logs.

### Automated Testing
Removed some tests that reference this dead code. 
 
### Regressions Considerations
As with modifying anything with Norman, it's not well documented and could have side effects. Most likely places would be in a system that has existed since v2.0.

pmatseykanets commented on pkg/api/norman/store/password/store.go:
Q: Maybe I missed something, but how password store related to `{cluster|project}loggings`?

JonCrowther replied:
From the PR that added the store https://github.com/rancher/rancher/pull/15968

> `password/store.go` contains generic logic to create secret for any field that is of type password. To add a schema that has password fields, it needs to be added to `pwdTypes`.

The only schemas in `password/store.go` are `clusterlogging` and `projectlogging`, so the store is a no-op if those CRDs are gone.

pmatseykanets replied:
👍 Probably worth noting this (that the norman password store is no longer used anywhere and is therefore removed) in the PR description.

JonCrowther replied:
Added!

JonCrowther commented:
This is failing integration tests because Shepherd copies the generated controllers, but not from this PR, so the build fails since `ClusterLogging` no longer exists. Seems like an oversight in design of Shepherd. Investigating how to cleanly resolve it.

crobby reviewed (dismissed):
No problems with what is here.  -1745 lines is lovely. 
Good luck sorting out the Shepherd part.

JonCrowther commented:
With the help of @caliskanugur, here is the method of updating shepherd:

1. Create a shepherd branch to use this branch's `pkg/apis` https://github.com/rancher/shepherd/pull/343
3. Update rancher/rancher PR to use this branch
4. Merge rancher/rancher PR
5. Update shepherd PR to use official rancher/rancher branch
6. Merge shepherd PR
7. Create a new rancher/rancher branch to point to official shepherd branch

I have updated go.mod with the right replace, and added a comment indicating why I made the change.

JonCrowther commented:
CI passed with the updated shepherd branch, but we don't want a fork in the rancher/rancher go.mod so we pushed past CI temporarily.
